subprojects/docs/src/docs/userguide/publishingOverview.adoc

```diff
@@ -0,0 +1,262 @@
+
+++++
+<sample dir="maven-publish/conditional-publishing" id="shorthandTasks" title="Defining your own shorthand tasks for publishing">
+    <sourcefile file="build.gradle" snippet="shorthand-tasks"/>
+</sample>
+++++
+
+This particular sample automatically handles the introduction or removal of the relevant publishing tasks by using api:org.gradle.api.tasks.TaskCollection#withType(java.lang.Class)[] with the api:org.gradle.api.publish.maven.tasks.PublishToMavenRepository[] task type. You can do the same with api:org.gradle.api.publish.ivy.tasks.PublishToIvyRepository[] if you're publishing to Ivy-compatible repositories.
+
+[[sec:configuring_publishing_tasks]]
+=== Configuring publishing tasks
+
+The publishing plugins create their non-aggregate tasks after the project has been evaluated, which means you cannot directly reference them from your build script. If you would like to configure any of these tasks, you should use deferred task configuration. This can be done in a number of ways via the project's `tasks` collection.
+
+For example, imagine you want to change where the `generatePomFileFor__PubName__Publication` tasks write their POM files. You can do this by using the api:org.gradle.api.tasks.TaskCollection#withType(java.lang.Class)[] method, as demonstrated by this sample:
+
+++++
+<sample dir="userguide/publishing/javaLibrary" id="deferred-configuration-publishing-task" title="Configuring a dynamically named task created by the publishing plugins">
+    <sourcefile file="build.gradle" snippet="configure-generate-task"/>
+</sample>
+++++
+
+The above sample uses a regular expression to extract the name of the publication from the name of the task. This is so that there is no conflict between the file paths of all the POM files that might be generated. If you only have one publication, then you don't have to worry about such conflicts since there will only be one POM file.
+
+=== Terminology
+
+[[glossary:artifact]]
+Artifact::
+A file produced by a build, such as a JAR, a ZIP distribution, or a native executable.
++
+Artifacts are typically designed to be used or consumed by users or other projects, or deployed to hosting systems. This is distinct from intermediate files that are created by a build as a step on the way to producing the end result. For example, the class files generated by compiling Java source are almost never an end product themselves, since they are typically packaged into JARs or some other type of archive.
+
+[[glossary:component]]
+Component::
+A predefined combination of one or more artifacts and a standard set of information about those artifacts. The concept exists in Gradle only as a part of publishing.
++
+Components are typically defined by plugins and provide a simple way to define a publication, as they already include the appropriate metadata. For example, the `java` component consists of the production JAR — produced by the `jar` task — as well as its dependency information.
```

Comment:
So if you publish a component, it will publish all the variants encompassed by that component? Also, I really, really don't want to overload terms. It's already hard enough to get one's head around modules, components, artifacts, and so on!

Reply:
Yes, if you publish a component, you publish all its variants.

Reply:
The terms aren't that hard if you attach examples to them:

Module: Guava 
Component: Guava 21
Variant: Guava 21 for Windows and JDK  9 (let's ignore that Guava has no such variants)
Artifact: guava-21-windows-jdk9.jar

Reply:
I think "component" is the one I struggle with the most as "a specific version of a module" is not a natural interpretation of the word. Nor does that interpretation seem to fit with the way it's used in publishing. The term "component" makes me think more of something "physical" like an artifact.

Anyway, I'm just thinking out loud here and trying to work out why I find this stuff confusing to some degree. I'll see what I can do.

Reply:
But that's what it is - it's a specific version of a piece of software. When you are publishing, you are always publishing a specific version of your software.